pkg/controller/storagecluster/reconcile.go

```diff
@@ -32,8 +32,13 @@ type ReconcileStrategy string
 // StorageClassProvisionerType is a string representing StorageClass Provisioner. E.g: aws-ebs
 type StorageClassProvisionerType string
 
-// ensureFunc which encapsulate all the 'ensure*' type functions
-type ensureFunc func(*ocsv1.StorageCluster, logr.Logger) error
+type resourceManager interface {
+	ensureCreated(*ReconcileStorageCluster, *ocsv1.StorageCluster, logr.Logger) error
+	ensureDeleted(*ReconcileStorageCluster, *ocsv1.StorageCluster, logr.Logger) error
```

Comment:
```suggestion
	ensureCreated(*ReconcileStorageCluster, *ocsv1.StorageCluster) error
	ensureDeleted(*ReconcileStorageCluster, *ocsv1.StorageCluster) error
```

`ReconcileStorageCluster` already has a logger which was being (duplicated and) passed around. We should make use of that and not encourage duplicating.

Reply:
I would like to take this up as the next task for refactor

Reply:
I think this should be covered in this PR or the initial design would seem flawed.
While we're re-factoring to enforce certain things, we shouldn't be enforcing a flawed design.

Reply:
I would still like to take it as a different PR as this is not just a case with this PR It has to removed from many places Other than this PR and will be a big task.

Reply:
Hi, @umangachapagain I have made changes as you suggested. I would request you to review it again as soon as possible. As it is a big change and takes a lot of time to resolve merge conflicts.